lib/data-types.js

```diff
@@ -39,24 +42,47 @@ var util = require('util')
  */
 
 var ABSTRACT = function(options) {
-
 };
 
 ABSTRACT.prototype.dialectTypes = '';
 
-ABSTRACT.prototype.toString = function() {
-  return this.toSql();
+ABSTRACT.prototype.toString = function(options) {
+  return this.toSql(options);
 };
 ABSTRACT.prototype.toSql = function() {
   return this.key;
 };
-ABSTRACT.prototype.warn = function(text) {
-  if (!warnings[text]) {
-    warnings[text] = true;
-    console.warn('>> WARNING:', text, '\n>> Check:', this.dialectTypes);
+ABSTRACT.warn = function(link, text) {
+  deprecate(text + '\n>> Check: ' + link);
+};
+ABSTRACT.prototype.stringify = function (value, options) {
+  if (this.$stringify) {
+    return this.$stringify(value, options);
+  }
+  return value;
+};
+
+ABSTRACT.inherits = function (Constructor) {
```

Comment:
Instead of calling `util.inherits` after each datatype, I added an inherits function to the abstract datatype, which inherits both prototype and static methods. Its also a utility method which creates the constructor for the new type, in cases where the constructor doesn't do anything interesting - saves a lot of copy-pasta